src/net_processing.cpp

```diff
@@ -2166,9 +2166,9 @@ void PeerManagerImpl::RelayTransaction(const Txid& txid, const Wtxid& wtxid)
         // in the announcement.
         if (tx_relay->m_next_inv_send_time == 0s) continue;
 
-        const auto gtxid{peer.m_wtxid_relay ? GenTxid{wtxid} : GenTxid{txid}};
-        if (!tx_relay->m_tx_inventory_known_filter.contains(gtxid.ToUint256())) {
-            tx_relay->m_tx_inventory_to_send.insert(gtxid);
+        const uint256& hash{peer.m_wtxid_relay ? wtxid.ToUint256() : txid.ToUint256()};
+        if (!tx_relay->m_tx_inventory_known_filter.contains(hash)) {
```

Comment:
Hm, is there a reason why `m_tx_inventory_known_filter` should contain sometimes-txids instead of wtxids?

Reply:
You're right. Might think about this some more but would be out of scope here anyway. Resolving!